README.md
# native-js-sdk
provide the native function call API of yunzai mobile platform for the third party


# How to user

## IN HTML

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8"/>
    <title>NativePage</title>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>
    <meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"/>
    <script type="text/javascript" src="sdk.js"></script>
</head>
<body>
<ol>
    <li>
        <button onclick="openWindow()">跳页面</button>
    </li>
    <li>
        <button onclick="setNavigationBarTitle()">设置标题</button>
    </li>
    <li>
        <button onclick="setNavigationBarRightItems()">设置按钮们</button>
    </li>
    <li>
        <button onclick="webShare()">设置分享</button>
    </li>
    <li>
        <button onclick="webCantShare()">设置不能分享</button>
    </li>
    <li>
        <div id="tokenasync"></div>
        <button onclick="getTokenAsync()">异步获取Token</button>
    </li>

    <li>
        <div id="tokensync"></div>
        <button onclick="getTokenSync()">同步获取Token</button>
    </li>

    <li>
        <div id="userAsync"></div>
        <button onclick="getUserAsync()">异步获取User</button>
    </li>

    <li>
        <div id="userSync"></div>
        <button onclick="getUserSync()">同步获取User</button>
    </li>

    <li>
        <div id="mediaCamera"></div>
        <button onclick="openMediaCamera()">开启你的摄像头</button>
    </li>

    <li>
        <div id="qrcode"></div>
        <button onclick="scanQrCodeAsync()">扫个二维码</button>
    </li>

    <li>
        <div id="userList"></div>
        <button onclick="chooseUser()">调用联系人</button>
    </li>

    <li>
        <div id="contactUserInfo"></div>
        <button onclick="getContactsInfoAsync()">获取联系人详细信息</button>
    </li>

    <li>
        <div id="photos"></div>
        <button onclick="uploadPhotoAsync()">上传图像们</button>
    </li>

    <li>
        <div id="deflocation"></div>
        <button onclick="userLocationAsync()">获取位置</button>
    </li>

    <li>
        <div id="wifimaclocation"></div>
        <button onclick="userLocationWifiAsync()">WIFI MAC地址验证</button>
    </li>

    <li>
        <div id="wifiinfo"></div>
        <button onclick="getWifiInfoAsync()">获取WIFI信息</button>
    </li>

    <li>
        <div id="wifimac"></div>
        <button onclick="getWifiMacAsync()">获取WIFIMAC地址</button>
    </li>
    <li>
        <div id="caiji"></div>
        <button onclick="faceLiveCollection()">人脸采集</button>
    </li>

    <li>
        <div id="duibi"></div>
        <button onclick="faceCompareAsync()">人脸对比</button>
    </li>

    <li>
        <div id="paywechat"></div>
        <button onclick="wechatPayAsync()">微信支付</button>
    </li>

    <li>
        <div id="payali"></div>
        <button onclick="aliPayAsync()">支付宝支付</button>
    </li>

    <li>
        <div id="readnumber"></div>
        <button onclick="openReadWithTimer()">计时阅读</button>
    </li>

    <li>
        <div id="deviceinfo"></div>
        <button onclick="getDeviceInfo()">设备信息</button>
    </li>


</ol>
<script type="text/javascript" src="main.bundle.js"></script></body>
<script>
    const device = new YzDevice();

    function getDeviceInfo() {
        device.getDeviceInfo().then((info) => {
            document.getElementById('deviceinfo').innerHTML = JSON.stringify(info);
        });
    }

    function openReadWithTimer() {
        device.openReadWithTimer({
            url: 'http://www.baidu.com',
            openType: "web",
            title: '百度',
            id: 'baidu',
            type: 'web'
        }).then((info) => {
            alert(info);
            document.getElementById('readnumber').innerHTML = JSON.stringify(info);
        })
    }

    function aliPayAsync() {
        device.aliPayAsync().then((info) => {
            document.getElementById('payali').innerHTML = JSON.stringify(info);
        })
    }

    function wechatPayAsync() {
        device.wechatPayAsync({
            "appid": "wx105c666575c89f9a",
            "partnerid": "1439124702",
            "prepayid": "wx04111944463057cc6b7954d61634639200",
            "package": "Sign=WXPay",
            "noncestr": "1cbfb3efb1de4fd186d85367f893d233",
            "timestamp": "1567567183",
            "sign": "FF993B12EE67DF76FD1E34FD26471454"
        }).then((info) => {
            document.getElementById('paywechat').innerHTML = JSON.stringify(info);
        })
    }

    function faceCompareAsync() {
        device.getUserAsync().then((user) => {
            device.faceCompareAsync({live: true, userId: user.userId}).then((data) => {
                document.getElementById("duibi").innerHTML = JSON.stringify(data);
            })
        });

    }

    function faceLiveCollection() {
        device.getUserAsync().then((user) => {
            device.faceCollectionAsync({live: true, userId: user.userId}).then((data) => {
                document.getElementById("caiji").innerHTML = JSON.stringify(data);
            })
        });
    }

    function getWifiMacAsync() {
        device.getWifiMacAsync().then((mac) => {
            document.getElementById("wifimac").innerHTML = mac;
        })
    }

    function getWifiInfoAsync() {
        device.getWifiInfoAsync().then((info) => {
            document.getElementById('wifiinfo').innerHTML = JSON.stringify(info);
        })
    }

    function userLocationWifiAsync() {
        device.userLocationWifiAsync({bssids: ['111.111.111.111', '222.222.222.222']}).then((status) => {
            document.getElementById('wifimaclocation').innerHTML = status;
        })
    }

    function userLocationAsync() {
        device.userLocationAsync().then((location) => {
            document.getElementById('deflocation').innerHTML = JSON.stringify(location);
        })
    }

    function uploadPhotoAsync() {
        device.uploadPhotoAsync({count: 100}).then((images) => {
            document.getElementById('photos').innerHTML = JSON.stringify(images);
        })
    }

    function chooseUser() {
        device.chooseContactsAsync({count: 1}).then((users) => {
            document.getElementById("userList").innerHTML = JSON.stringify(users);
        })
    }

    function getContactsInfoAsync() {
        device.chooseContactsAsync({count: 1}).then((users) => {
            device.getContactsInfoAsync({targetUserId: users[0].id}).then((userInfo) => {
                document.getElementById('contactUserInfo').innerHTML = JSON.stringify(userInfo);
            })
        })
    }

    function scanQrCodeAsync() {
        device.scanQrCodeAsync().then((code) => {
            document.getElementById("qrcode").innerHTML = code;
        });
    }


    function openMediaCamera() {
        device.openMediaCameraAsync().then((camera) => {
            document.getElementById("mediaCamera").innerHTML = camera;
        })
    }

    function getUserAsync() {
        device.getUserAsync().then((user) => {
            document.getElementById('userAsync').innerHTML = JSON.stringify(user);
        })
    }

    function getUserSync() {
        let user = device.getUserSync();
        document.getElementById('userSync').innerHTML = JSON.stringify(user);
    }

    function getTokenSync() {
        let token = device.getTokenSync();
        document.getElementById('tokensync').innerHTML = JSON.stringify(token);
    }

    function getTokenAsync() {
        device.getTokenAsync({refresh: true}).then((token) => {
            document.getElementById('tokenasync').innerHTML = JSON.stringify(token);
        });
    };

    function webCantShare() {
        device.setWebCanshare({shareFlag: false});
    }

    function webShare() {
        device.setWebCanshare({shareFlag: true});
    }

    function openWindow() {
        device.openWindow({
            type: "url",
            url: "http://www.baidu.com",
            shareFlag: true,
            success: function (msg) {
                alert(JSON.stringify(msg));
            },
            fail: function (msg) {
                alert(JSON.stringify(msg));
            },
            complete: function (msg) {
                alert(JSON.stringify(msg));
            }
        });
    }

    function setNavigationBarTitle() {
        device.setNavigationBarTitle({
            title: "哈哈哈哈哈",
            success: function (msg) {
                alert(JSON.stringify(msg));
            },
            fail: function (msg) {
                alert(JSON.stringify(msg));
            },
            complete: function (msg) {
                alert(JSON.stringify(msg));
            }
        });
    }

    function setNavigationBarRightItems() {
        device.setNavigationBarRightItems({
            showMenu: true,
            shareFlag: true,
            items: [
                {
                    type: "icon", //显示类型：图标，文字
                    title: "",
                    icon: "edit", // 目前支持图标按钮：add添加, edit编辑, delete删除, search搜索, done完成, send发送
                    clickId: "edit" //点击Id
                },
                {
                    type: "icon", //显示类型：图标，文字
                    title: "",
                    icon: "delete", // 目前支持图标按钮：add添加, edit编辑, delete删除, search搜索, done完成, send发送
                    clickId: "123123" //点击Id
                },
                {
                    type: "icon", //显示类型：图标，文字
                    title: "",
                    icon: "search", // 目前支持图标按钮：add添加, edit编辑, delete删除, search搜索, done完成, send发送
                    clickId: "hahah" //点击Id
                },
                {
                    type: "icon", //显示类型：图标，文字
                    title: "",
                    icon: "done", // 目前支持图标按钮：add添加, edit编辑, delete删除, search搜索, done完成, send发送
                    clickId: "hahah" //点击Id
                },
                {
                    type: "icon", //显示类型：图标，文字
                    title: "",
                    icon: "send", // 目前支持图标按钮：add添加, edit编辑, delete删除, search搜索, done完成, send发送
                    clickId: "hahah" //点击Id
                }
            ],
            onClick: function (clickId) {
                alert(clickId + "按钮1被点击了");
            }
        });
    }
</script>
</html>
```

## IN TYPESCRIPT

```typescript
import * as yz from 'yz-natice-js-sdk';
const device = new YzDevice();
device.***
```


CHANGELOG.md
# 0.1.17 2020-04-13
## feature
1.加入浏览器下载函数
```javascript
  device.downloadByBrowser({
            url: 'http://www.baidu.com',
            title: '谷歌',
            success: function () {
                alert("成功");
            },
            fail: function (error) {
                alert(error);
            },
            complete: function () {
                alert("函数完成后回调")
            }
        }).then((data) => {
            console.log(data);
        });
```

# 0.1.16 2020-04-13
## fix
1.修复user获取为字符串,现在直接返回对象  
2.修复token获取为字符串,现在直接返回token对象

## feature
1.加入 浏览器打开函数

```javascript
 function openBrowser() {
        device.fileBrowser({
            url: 'http://www.baidu.com',
            title: '谷歌',
            success: function (data) {
                alert(data);
            },
            fail: function (data) {
                alert(data);
            },
            complete: function () {
                alert("函数完成后回调")
            }
        }).then((data) => {
            console.log(data);
        });
    }
```

# 0.1.15 2020-04-7
1.修复http请求后不返回Promise问题
# 0.1.14 2020-04-7
1.修复user不是Object问题
# 0.1.13 2020-02-14
## fix
1.跨域请求许可
# 0.1.12 2020-02-14
## fix
1.用户认证
# 0.1.5 2020-02-05
## feature
1.加入微信支持
2.加入微信User/Location/Qrcode
# 0.1.4 2019-12-25
## fix
1.修复查看联系人详情function
# 0.1.2 2019-12-17
## fix
1.修复index.html调用函数webCanshare 为 webCanShare,解决调用不到函数的问题
# 0.1.1 2019-12-11
## feature
<ol>
    <li>
        <button onclick="openWindow()">跳页面</button>
    </li>
    <li>
        <button onclick="setNavigationBarTitle()">设置标题</button>
    </li>
    <li>
        <button onclick="setNavigationBarRightItems()">设置按钮们</button>
    </li>
    <li>
        <button onclick="webShare()">设置分享</button>
    </li>
    <li>
        <button onclick="webCantShare()">设置不能分享</button>
    </li>
    <li>
        <div id="tokenasync"></div>
        <button onclick="getTokenAsync()">异步获取Token</button>
    </li>
    <li>
        <div id="tokensync"></div>
        <button onclick="getTokenSync()">同步获取Token</button>
    </li>
    <li>
        <div id="userAsync"></div>
        <button onclick="getUserAsync()">异步获取User</button>
    </li>
    <li>
        <div id="userSync"></div>
        <button onclick="getUserSync()">同步获取User</button>
    </li>
    <li>
        <div id="mediaCamera"></div>
        <button onclick="openMediaCamera()">开启你的摄像头</button>
    </li>
    <li>
        <div id="qrcode"></div>
        <button onclick="scanQrCodeAsync()">扫个二维码</button>
    </li>
    <li>
        <div id="userList"></div>
        <button onclick="chooseUser()">调用联系人</button>
    </li>
    <li>
        <div id="contactUserInfo"></div>
        <button onclick="getContactsInfoAsync()">获取联系人详细信息</button>
    </li>
    <li>
        <div id="photos"></div>
        <button onclick="uploadPhotoAsync()">上传图像们</button>
    </li>
    <li>
        <div id="deflocation"></div>
        <button onclick="userLocationAsync()">获取位置</button>
    </li>
    <li>
        <div id="wifimaclocation"></div>
        <button onclick="userLocationWifiAsync()">WIFI MAC地址验证</button>
    </li>
    <li>
        <div id="wifiinfo"></div>
        <button onclick="getWifiInfoAsync()">获取WIFI信息</button>
    </li>
    <li>
        <div id="wifimac"></div>
        <button onclick="getWifiMacAsync()">获取WIFIMAC地址</button>
    </li>
    <li>
        <div id="caiji"></div>
        <button onclick="faceLiveCollection()">人脸采集</button>
    </li>
    <li>
        <div id="duibi"></div>
        <button onclick="faceCompareAsync()">人脸对比</button>
    </li>
    <li>
        <div id="paywechat"></div>
        <button onclick="wechatPayAsync()">微信支付</button>
    </li>
    <li>
        <div id="payali"></div>
        <button onclick="aliPayAsync()">支付宝支付</button>
    </li>
    <li>
        <div id="readnumber"></div>
        <button onclick="openReadWithTimer()">计时阅读</button>
    </li>
    <li>
        <div id="deviceinfo"></div>
        <button onclick="getDeviceInfo()">设备信息</button>
    </li>
</ol>


src/sdk.js
(function() {
    var src = "//cdn.jsdelivr.net/npm/eruda";
    if (
        !/eruda=true/.test(window.location) &&
        localStorage.getItem("active-eruda") != "true"
    )
        return;
    document.write("<scr" + 'ipt src="' + src + '"></scr' + "ipt>");
    document.write("<scr" + "ipt>eruda.init();</scr" + "ipt>");
})();


var bridge = {
    default: this, call: function (b, a, c) {
        var e = "";
        "function" == typeof a && (c = a, a = {});
        a = {data: void 0 === a ? null : a};
        if ("function" == typeof c) {
            var g = "dscb" + window.dscb++;
            window[g] = c;
            a._dscbstub = g
        }
        a = JSON.stringify(a);
        if (window._dsbridge) e = _dsbridge.call(b, a); else if (window._dswk || -1 != navigator.userAgent.indexOf("_dsbridge")) e = prompt("_dsbridge=" + b, a);
        return JSON.parse(e || "{}").data
    }, register: function (b, a, c) {
        c = c ? window._dsaf : window._dsf;
        window._dsInit || (window._dsInit = !0, setTimeout(function () {
                bridge.call("_dsb.dsinit")
            },
            0));
        "object" == typeof a ? c._obs[b] = a : c[b] = a
    }, registerAsyn: function (b, a) {
        this.register(b, a, !0)
    }, hasNativeMethod: function (b, a) {
        return this.call("_dsb.hasNativeMethod", {name: b, type: a || "all"})
    }, disableJavascriptDialogBlock: function (b) {
        this.call("_dsb.disableJavascriptDialogBlock", {disable: !1 !== b})
    }
};
!function () {
    if (!window._dsf) {
        var b = {
            _dsf: {_obs: {}}, _dsaf: {_obs: {}}, dscb: 0, dsBridge: bridge, close: function () {
                bridge.call("_dsb.closePage")
            }, _handleMessageFromNative: function (a) {
                var e = JSON.parse(a.data), b = {id: a.callbackId, complete: !0}, c = this._dsf[a.method],
                    d = this._dsaf[a.method], h = function (a, c) {
                        b.data = a.apply(c, e);
                        bridge.call("_dsb.returnValue", b)
                    }, k = function (a, c) {
                        e.push(function (a, c) {
                            b.data = a;
                            b.complete = !1 !== c;
                            bridge.call("_dsb.returnValue", b)
                        });
                        a.apply(c, e)
                    };
                if (c) h(c, this._dsf); else if (d) k(d, this._dsaf);
                else if (c = a.method.split("."), !(2 > c.length)) {
                    a = c.pop();
                    var c = c.join("."), d = this._dsf._obs, d = d[c] || {}, f = d[a];
                    f && "function" == typeof f ? h(f, d) : (d = this._dsaf._obs, d = d[c] || {}, (f = d[a]) && "function" == typeof f && k(f, d))
                }
            }
        }, a;
        for (a in b) window[a] = b[a];
        bridge.register("_hasJavascriptMethod", function (a, b) {
            b = a.split(".");
            if (2 > b.length) return !(!_dsf[b] && !_dsaf[b]);
            a = b.pop();
            b = b.join(".");
            return (b = _dsf._obs[b] || _dsaf._obs[b]) && !!b[a]
        })
    }
}();

var yz = {
    config: {
        DEBUG: false,
        version: 6
    },
    // 1.界面
    // 1.1导航
    navigationOpenWindow: function (option) {
        if (option.shareFlag == null || option.shareFlag == undefined) {
            option.shareFlag = false
        }
        if (option.nBack == null || option.nBack == undefined) {
            option.nBack = false
        }
        if (option.type == null || option.type == undefined) {
            option.type = "url"
        }
        if (option.type === "url") {
            if (!option.url) {
                throw Error('lost url param')
            }
        }
        yz._private.call('JsV6.navigationOpenWindow', {
            type: option.type,
            applicationId: option.applicationId,
            url: option.url,
            shareFlag: option.shareFlag,
            nBack: option.nBack
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 1.2设置导航条
    setNavigationBarTitle: function (option) {
        if (!option.title) {
            throw Error('lost title param')
        }
        yz._private.call('JsV6.setNavigationBarTitle', {
            title: option.title
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 设置右部按钮
    setNavigationBarRightItems: function (option) {
        if (option.showMenu == null || option.showMenu == undefined) {
            option.showMenu = true
        }
        if (option.shareFlag == null || option.shareFlag == undefined) {
            option.shareFlag = false
        }
        if (!option.items) {
            throw Error('lost items param')
        }
        if (!option.onClick && 'function' !== typeof option.onClick) {
            throw Error('lost onClick function')
        }
        yz._private.call('JsV6.setNavigationBarRightItems', {
            showMenu: option.showMenu,
            shareFlag: option.shareFlag,
            items: option.items,
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        });
        bridge.register('navigationBarRightItemOnClick', function (title) {
            option.onClick(title)
        });
    },
    // 1.3网页分享
    setWebCanShare: function (option) {
        if (option.shareFlag == null || option.shareFlag == undefined) {
            option.shareFlag = false
        }
        yz._private.call('JsV6.setWebCanShare', {
            shareFlag: option.shareFlag
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        });
    },
    // 2.用户信息
    // 2.1异步Token
    getToken: function (option) {
        if (option.refresh == null || option.refresh == undefined) {
            option.refresh = false
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.getToken', {
            refresh: option.refresh
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        });
    },
    // 同步获取Token
    getTokenSync: function () {
        var resultData = yz._private.callSync('JsV6.getTokenSync', {});
        if (resultData.status) {
            return resultData.data;
        } else {
            throw Error(resultData.msg)
        }
    },
    // 2.2同步获取用户信息
    getUserInfoSync: function () {
        var resultData = yz._private.callSync('JsV6.getUserInfoSync', {});
        if (resultData.status) {
            return resultData.data;
        } else {
            throw Error(resultData.msg)
        }
    },
    // 异步获取用户信息
    getUserInfo: function (option) {
        if (!option) {
            throw Error('lost option')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.getUserInfo', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 3.扫码
    scanCode: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.scanCode', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 4.1选择联系人
    chooseContacts: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        if (!option.count) {
            option.count = 0
        }
        yz._private.call('JsV6.chooseContacts', {
            count: option.count
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 4.2联系人详情
    ContactDetails: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        if (!option.targetUserId) {
            throw Error('lost targetUserId param')
        }
        yz._private.call('JsV6.ContactDetails', {
            targetUserId: option.targetUserId
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 5.图片选择
    selectImage: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        if (!option.count) {
            option.count = 1
        }
        if (!option.sourceType) {
            option.sourceType = [
                'album', 'camera'
            ]
        }
        yz._private.call('JsV6.selectImage', {
            count: option.count,
            sourceType: option.sourceType
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 6.上传图片
    uploadImage: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        if (!option.count) {
            option.count = 1
        }
        if (!option.sourceType) {
            option.sourceType = [
                'album', 'camera'
            ]
        }
        yz._private.call('JsV6.uploadImage', {
            count: option.count,
            sourceType: option.sourceType
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 7.上传视频
    uploadMovie: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        if (!option.limitMinute) {
            option.limitMinute = 3
        }
        yz._private.call('JsV6.uploadMovie', {
            limitMinute: option.limitMinute
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 8.顶部颜色
    setNavigationBarColor: function (option) {
        if (!option.frontColor) {
            throw Error('lost frontColor param')
        }
        if (!option.backgroundColor) {
            throw Error('lost backgroundColor param')
        }
        yz._private.call('JsV6.setNavigationBarColor', {
            frontColor: option.frontColor,
            backgroundColor: option.backgroundColor
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 9.播放器
    mediaPlayer: function (option) {
        if (!option.url) {
            throw Error('lost url param')
        }
        if (!option.skipTime) {
            option.skipTime = 0
        }
        if (option.title === undefined || option.title === null) {
            option.title = ''
        }
        if (option.content === undefined || option.content === null) {
            option.content = ''
        }
        yz._private.call('JsV6.mediaPlayer', {
            url: option.url,
            skipTime: option.skipTime,
            title: option.title,
            content: option.content
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 10.定位获取经纬度
    userLocation: function (option) {
        if (!option) {
            throw Error('lost option')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.userLocation', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 11.Ap定位
    userLocationContainAp: function (option) {
        if (!option.bssids) {
            throw Error('lost bssids param')
        }
        yz._private.call('JsV6.userLocationContainAp', {
            bssids: option.bssids
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 12.1获取已连接Wifi信息
    getWifiInfo: function (option) {
        if (!option) {
            throw Error('lost option')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.getWifiInfo', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 12.2获取已连接Wifi的Mac地址
    getMac: function (option) {
        if (!option) {
            throw Error('lost option')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.getMac', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 13.打开相机
    openCamera: function (option) {
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.openCamera', {}, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 14.1获取微信AppId
    getWXAppId: function (option) {
        var resultData = yz._private.callSync('JsV6.getWXAppId', {});
        if (resultData.status) {
            return resultData.data;
        } else {
            throw Error(resultData.msg)
        }
    },
    // 14.2微信支付
    wechatPay: function (option) {
        if (!option.payInfo) {
            throw Error('lost payInfo param')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.wechatPay', {
            payInfo: option.payInfo
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 15.支付宝支付
    aliPay: function (option) {
        if (!option.payInfo) {
            throw Error('lost payInfo param')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.aliPay', {
            payInfo: option.payInfo
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 16.人脸采集
    faceLiveCollection: function (option) {
        if (!option.live) {
            throw Error('lost live param')
        }
        if (!option.userId) {
            throw Error('lost userId param')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.faceLiveCollection', {
            live: option.live,
            userId: option.userId
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 17.人脸对比
    faceCompare: function (option) {
        if (!option.userId) {
            throw Error('lost userId param')
        }
        if (!option.success) {
            throw Error('lost success function')
        }
        yz._private.call('JsV6.faceCompare', {
            userId: option.userId
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 18.综合训练读物时间统计时间
    openReadWithTimer: function (option) {
        if (!option.url) {
            throw Error('lost url param')
        }
        if (!option.openType) {
            throw Error('lost openType param')
        }
        if (!option.title) {
            throw Error('lost title param')
        }
        if (!option.id) {
            throw Error('lost id param')
        }
        if (!option.type) {
            throw Error('lost type param')
        }
        yz._private.call('JsV6.openReadWithTimer', {
            url: option.url,
            openType: option.openType,
            title: option.title,
            id: option.id,
            type: option.type
        }, function (resultData) {
            yz._private.handlerResult(resultData, option)
        })
    },
    // 注册监听
    listener: {
        // 下一个window返回上一个window触发
        nBack: function (functionF) {
            yz._private.register("nBack", functionF);
        }
    },
    _private: {
        register: function (functionName, functionF) {
            if (typeof functionF !== 'function') {
                throw Error('param isNot function')
            }
            if (yz.config.DEBUG) {
                console.log('----- 注册监听 -----\n functionName:' + functionName);
            }
            bridge.register(functionName, functionF);
        },
        callSync: function (functionName, param) {
            var paramString = JSON.stringify(param);
            if (yz.config.DEBUG) {
                console.log('----- 调用原生方法 -----\n functionName:' + functionName + '\n param:' + paramString);
            }
            var data = bridge.call(functionName, paramString);
            if (yz.config.DEBUG) {
                console.log('+++++ 原生方法结果 +++++\n functionName:' + functionName + '\n result:' + data);
            }
            return JSON.parse(data);
        },
        call: function (functionName, param, callBack) {
            var paramString = JSON.stringify(param);
            if (yz.config.DEBUG) {
                console.log('----- 调用原生方法async -----\n functionName:' + functionName + '\n param:' + paramString);
            }
            bridge.call(functionName, paramString, function (data) {
                if (yz.config.DEBUG) {
                    console.log('+++++ 原生方法结果async +++++\n functionName:' + functionName + '\n result:' + data);
                }
                callBack(JSON.parse(data));
            });
        },
        handlerResult: function (resultData, option) {
            if (resultData.status) {
                if (option.success && 'function' === typeof option.success) {
                    option.success(resultData.data)
                }
            } else {
                if (option.fail && 'function' === typeof option.fail) {
                    option.fail(resultData.msg)
                }
            }
            if (option.complete && 'function' === typeof option.complete) {
                option.complete()
            }
        }
    }
};
